File: 백준/Silver/1735. 분수 합/분수 합.c
#include <stdio.h>
#include <stdlib.h>

long gcd(long x, long y) {
    return x % y == 0 ? y : gcd(y, x % y);
}

int main(void) {
    int** num = (int**)malloc(sizeof(int*) * 2);
    long numer, denom;
    for (int i = 0; i < 2; i++) {
        num[i] = (int*)malloc(sizeof(int) * 2);
        scanf("%d %d", &num[i][0], &num[i][1]);
    }
    numer = num[0][0] * num[1][1] + num[0][1] * num[1][0];
    denom = num[0][1] * num[1][1];
    printf("%ld %ld", numer / gcd(numer, denom), denom / gcd(numer, denom));
    return 0;
}

File: 백준/Silver/1735. 분수 합/분수 합.py
import sys
num=[]
def gcd(x,y):
    return y if x%y==0 else gcd(y,x%y)
for i in range(2):
    num+=[list(map(int,sys.stdin.readline().split()))]
numer=num[0][0]*num[1][1]+num[0][1]*num[1][0]
denom=num[0][1]*num[1][1]
print(numer//gcd(numer,denom), denom//gcd(numer,denom))